File: SchoolWebApp-API/Project.Core/Entities/Staff/StaffSubject.cs
using SchoolWebApp.Core.Entities.Academics;
using SchoolWebApp.Core.Entities.Class;
using SchoolWebApp.Core.Entities.Shared;

namespace SchoolWebApp.Core.Entities.Staff
{
    public class StaffSubject : Base
    {
        public string? Description { get; set; }
        public int StaffDetailsId { get; set; }
        public StaffDetails? StaffDetails { get; set; }
        public int SubjectId { get; set; }
        public Subject? Subject { get; set; }
        public int SchoolClassId { get; set; }
        public SchoolClass? SchoolClass { get; set; }
    }
}

File: SchoolWebApp-Client/src/app/staff/models/staff-subject.ts
import { ResourceModel } from "@/core/models/ResourceModel";
import { StaffDetails } from "./staff-details";
import { Subject } from "@/academics/models/subject";
import { SchoolClass } from "@/class/models/school-class";

export class StaffSubject extends ResourceModel<StaffSubject> {
    public staffDetailsId?: number;
    public staffDetails?: StaffDetails;
    
    public description?: string;

    public subjectId?: number;
    public subject?: Subject;
    public schoolClassId?: number;
    public schoolClass?: SchoolClass;

    constructor(model?: Partial<StaffSubject>) {
        super(model);
    }
}